pkg/processor/runtime/context.go

```diff
@@ -17,21 +17,77 @@ limitations under the License.
 package runtime
 
 import (
+	"net/url"
+
+	"github.com/iguazio/v3io-go-http"
 	"github.com/nuclio/nuclio-sdk"
-	"github.com/nuclio/nuclio/pkg/v3ioclient"
+	"github.com/pkg/errors"
 )
 
-func newContext(logger nuclio.Logger, configuration *Configuration) *nuclio.Context {
+func newContext(parentLogger nuclio.Logger, configuration *Configuration) (*nuclio.Context, error) {
 	newContext := &nuclio.Context{
-		Logger: logger,
+		Logger:      parentLogger,
+		DataBinding: map[string]nuclio.DataBinding{},
 	}
 
 	// create v3io context if applicable
-	for _, dataBinding := range configuration.DataBindings {
+	for dataBindingName, dataBinding := range configuration.DataBindings {
 		if dataBinding.Class == "v3io" {
-			newContext.DataBinding = v3ioclient.NewV3ioClient(logger, dataBinding.URL)
+
+			// create a container object that can be used by the event handlers
+			container, err := createV3ioDataBinding(parentLogger, dataBinding.Url)
+			if err != nil {
+				return nil, errors.Wrapf(err, "Failed to create v3io client for %s", dataBinding.Url)
+			}
+
+			newContext.DataBinding[dataBindingName] = container
 		}
 	}
 
-	return newContext
+	return newContext, nil
+}
+
+func createV3ioDataBinding(parentLogger nuclio.Logger, url string) (*v3io.Container, error) {
+
+	// parse the URL to get address and container ID
+	addr, containerAlias, err := parseURL(url)
+
+	// create context
+	context, err := v3io.NewContext(parentLogger, addr)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create client")
+	}
+
+	// create session
+	session, err := context.NewSession("", "", "nuclio")
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create session")
+	}
+
+	// create the container
+	container, err := session.NewContainer(containerAlias)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create container")
+	}
+
+	return container, nil
+}
+
+func parseURL(rawURL string) (addr string, containerAlias string, err error) {
+	parsedURL, err := url.Parse(rawURL)
```

Comment:
I don't think I ever seen `url.Parse` fail. It'll be nice to have the `rawURL` in the error message.

Reply:
:+1: about error message